sparselsh/lsh.py

```diff
@@ -192,51 +193,56 @@ def _as_np_array(self, serial_or_sparse):
                 raise
         else:
             raise TypeError("the input data is not supported")
-
-    def index(self, input_point, extra_data=None):
-        """ Index a single input point by adding it to the selected storage.
-
+    
+    def index(self, input_points, extra_data=None):
+        """ Index input points by adding them to the selected storage.
+        
         If `extra_data` is provided, it will become the value of the dictionary
         {input_point: extra_data}, which in turn will become the value of the
         hash table.
-
-        :param input_point:
-            A sparse CSR matrix. The dimension needs to be 1 * `input_dim`.
+        
+        :param input_points:
+            A sparse CSR matrix. The dimension needs to be N x `input_dim`, N>0.
         :param extra_data:
-            (optional) A value to associate with the point. Commonly this is
-            a target/class-value of some type.
+            (optional) A list of values to associate with the points. Commonly
+            this is a target/class-value of some type.
         """
-
-        assert sparse.issparse(input_point), "input_point needs to be sparse"
-
-        # NOTE: there was a bug with 0-equal extra_data
-        # we need to allow blank extra_data if it's provided
-        if not isinstance(extra_data, type(None)):
-            # NOTE: needs to be tuple so it's set-hashable
-            value = (input_point, extra_data)
-        else:
-            value = input_point
-
+        
+        assert sparse.issparse(input_points), "input_points needs to be sparse"
+        
```

Comment:
maybe we can assert that the shape of input_points matches that of extra_data (if it's provided) as well?

Reply:
I will check the extra_data shape assertion but I am unsure what the extra_data might be and its format. Could you give me some more info?

Reply:
Extra data is a class or some additional metadata to be kept along with the input data. Since the LSH index can only store the input points, this lets programmers index some points and be able to also retrieve associated metadata. An example use case would be a data set of names of home buyers and extra_data is the description of the property they owned. Doing a lookup on a new name will let you get similar names along with the properties associated with those names. I think we just want to make sure extra_data has the same length as input_points if it's not None. But like I said, completely optional!

Reply:
Assuming extra_data is a list of values, this check has been implemented.